### DISCUSSION TITLE
Call for code documentation

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#6890] [Ideas]
It is quite frustrating at times to dig into filament's source code as pretty much nothing is documented. 
It would make A LOT of sense to have classes and functions have a DocBlock that explains what it is about. 
Also adding snippets like @param @return ee and so on would help immensely when working with an IDE.
This would have the potential to reduce the amount of support requests.

**Comment by danharrin** [comment]
I understand wanting descriptions, but `@param` and `@return` are useless as we already use PHP types extensively which do the same thing.

**Comment by OvalMedia** [comment]
So, is there a path to inline documentation?

**Reply by danharrin** [reply]
If you're willing to put the hours in... but I don't have the time at the moment.
